packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -43,75 +54,113 @@ export class LoginPage {
 	private async pageSettled(): Promise< void > {
 		// Needs to be `networkidle`, otherwise switching accounts will fail.
 		await this.page.waitForLoadState( 'networkidle' );
-		const container = await this.page.waitForSelector( selectors.loginContainer );
-		// The login container can fade in or shift, so wait for that to complete.
-		await container.waitForElementState( 'stable' );
+	}
+
+	/**
+	 * Navigates to the /log-in endpoint.
+	 */
+	async visit(): Promise< void > {
+		await this.page.goto( getCalypsoURL( 'log-in' ), { waitUntil: 'networkidle' } );
 	}
 
 	/**
 	 * Switch account if an user is already logged in.
 	 */
 	private async switchAccount(): Promise< void > {
-		// By default, log out of the existing account (even if test steps end up logging back in).
-		const alreadyLoggedIn = await this.page.$( selectors.changeAccountButton );
+		await this.pageSettled();
+
+		// Change account button takes some time to fade into view if an account
+		// is already logged in.
+		const alreadyLoggedIn = await this.page.isVisible( selectors.continueAsUser );
 		if ( alreadyLoggedIn ) {
-			console.log( 'already logged in, selecting "change account' );
+			const elementHandle = await this.page.waitForSelector( selectors.continueAsUser );
+			await elementHandle.waitForElementState( 'stable' );
```

Comment:
This part addresses #56400 by waiting for the `continueAsUser` div to finish fading in before proceeding.